test/services/metadata_templates/update_service_test.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+
+module MetadataTemplates
+  class UpdateServiceTest < ActiveSupport::TestCase
+    def setup
+      @user = users(:john_doe)
+      @metadata_template = metadata_templates(:valid_metadata_template)
+    end
+
+    test 'updates metadata template with valid params' do
+      valid_params = { name: 'new-metadata-template-name', description: 'new-metadata-template-description' }
+
+      assert_changes -> { [@metadata_template.name, @metadata_template.description] },
+                     to: %w[new-metadata-template-name new-metadata-template-description] do
+        MetadataTemplates::UpdateService.new(@user, @metadata_template, valid_params).execute
+      end
+    end
+
+    test 'failes to updates metadata template with invalid params' do
+      invalid_params = { fields: nil }
+
+      assert_no_changes -> { @metadata_template.reload.fields } do
+        MetadataTemplates::UpdateService.new(@user, @metadata_template, invalid_params).execute
+      end
+      assert_includes @metadata_template.errors[:fields],
+                      'value at root is not an array'
+    end
+
+    test 'fails to update metadata template with numerical fields' do
+      invalid_params = { fields: [1, 2, 3, 4] }
+
+      assert_no_changes -> { @metadata_template.reload.fields } do
+        MetadataTemplates::UpdateService.new(@user, @metadata_template, invalid_params).execute
+      end
```

Comment:
We should as an assert_includes like the test above but for a message indicating that the items aren't a string value

Reply:
Updated in f0167f8b2326862dfb7cbec67132061548987d7c

Reply:
Updated in f0167f8b2326862dfb7cbec67132061548987d7c